<h1>ACTIVITY 1 : PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>
<h3>Analyst : Mark John Well D. De Castro</h3>

In [1]:
pip install numpy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.2.1 -> 25.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.2.1 -> 25.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [3]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')
from scipy.stats import f_oneway

<h2>Converting Datasets to DataFrames</h2>
<br>
<h2>Requirements</h2>
<ul>
    <li>DataFrame name</li>
    <li>File Path</li>
</ul>

In [4]:
penguins = pd.read_csv("datasets\\penguins.csv")

In [5]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h1>DESCRIPTIVE ANALYTICS</h1>

In [6]:
penguins.describe()

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      342.000000     342.000000         342.000000   
mean   172.500000       43.921930      17.151170         200.915205   
std     99.448479        5.459584       1.974793          14.061714   
min      1.000000       32.100000      13.100000         172.000000   
25%     86.750000       39.225000      15.600000         190.000000   
50%    172.500000       44.450000      17.300000         197.000000   
75%    258.250000       48.500000      18.700000         213.000000   
max    344.000000       59.600000      21.500000         231.000000   

       body_mass_g         year  
count   342.000000   344.000000  
mean   4201.754386  2008.029070  
std     801.954536     0.818356  
min    2700.000000  2007.000000  
25%    3550.000000  2007.000000  
50%    4050.000000  2008.000000  
75%    4750.000000  2009.000000  
max    6300.000000  2009.000000

In [7]:
penguins.describe()

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      342.000000     342.000000         342.000000   
mean   172.500000       43.921930      17.151170         200.915205   
std     99.448479        5.459584       1.974793          14.061714   
min      1.000000       32.100000      13.100000         172.000000   
25%     86.750000       39.225000      15.600000         190.000000   
50%    172.500000       44.450000      17.300000         197.000000   
75%    258.250000       48.500000      18.700000         213.000000   
max    344.000000       59.600000      21.500000         231.000000   

       body_mass_g         year  
count   342.000000   344.000000  
mean   4201.754386  2008.029070  
std     801.954536     0.818356  
min    2700.000000  2007.000000  
25%    3550.000000  2007.000000  
50%    4050.000000  2008.000000  
75%    4750.000000  2009.000000  
max    6300.000000  2009.000000

<h2>Checking for Null Values</h2>

In [8]:
penguins.isnull()

rowid  species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    False    False   False           False          False              False   
1    False    False   False           False          False              False   
2    False    False   False           False          False              False   
3    False    False   False            True           True               True   
4    False    False   False           False          False              False   
..     ...      ...     ...             ...            ...                ...   
339  False    False   False           False          False              False   
340  False    False   False           False          False              False   
341  False    False   False           False          False              False   
342  False    False   False           False          False              False   
343  False    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 9 columns]

<h2>Summarizing Null Values</h2>

In [9]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [10]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [11]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h2>Filtering a Datasets</h2>

In [12]:
penguins[penguins['species'] == 'Adelie']

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
147    148  Adelie      Dream            36.6           18.4   
148    149  Adelie      Dream            36.0           17.8   
149    150  Adelie      Dream            37.8           18.1   
150    151  Adelie      Dream            36.0           17.1   
151    152  Adelie      Dream            41.5           18.5   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
147              184.0       3475.0  female  2009  
148              195.0       3450.0  female  2009  
149              193.0       3750.0    male  2009  
150              187.0       3700.0  female  2009  
151              201.0       4000.0    male  2009  

[152 rows x 9 columns]

In [13]:
penguins[penguins['species'] == 'Chinstrap']

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [14]:
penguins[penguins['species'] == 'Gentoo']

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152    153  Gentoo  Biscoe            46.1           13.2              211.0   
153    154  Gentoo  Biscoe            50.0           16.3              230.0   
154    155  Gentoo  Biscoe            48.7           14.1              210.0   
155    156  Gentoo  Biscoe            50.0           15.2              218.0   
156    157  Gentoo  Biscoe            47.6           14.5              215.0   
..     ...     ...     ...             ...            ...                ...   
271    272  Gentoo  Biscoe             NaN            NaN                NaN   
272    273  Gentoo  Biscoe            46.8           14.3              215.0   
273    274  Gentoo  Biscoe            50.4           15.7              222.0   
274    275  Gentoo  Biscoe            45.2           14.8              212.0   
275    276  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 9 columns]

<h2>Assigning variable Name</h2>

In [15]:
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_df

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
147    148  Adelie      Dream            36.6           18.4   
148    149  Adelie      Dream            36.0           17.8   
149    150  Adelie      Dream            37.8           18.1   
150    151  Adelie      Dream            36.0           17.1   
151    152  Adelie      Dream            41.5           18.5   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
147              184.0       3475.0  female  2009  
148              195.0       3450.0  female  2009  
149              193.0       3750.0    male  2009  
150              187.0       3700.0  female  2009  
151              201.0       4000.0    male  2009  

[152 rows x 9 columns]

In [16]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [17]:
chinstrap_df = penguins[penguins['species'] == 'Gentoo']
chinstrap_df

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152    153  Gentoo  Biscoe            46.1           13.2              211.0   
153    154  Gentoo  Biscoe            50.0           16.3              230.0   
154    155  Gentoo  Biscoe            48.7           14.1              210.0   
155    156  Gentoo  Biscoe            50.0           15.2              218.0   
156    157  Gentoo  Biscoe            47.6           14.5              215.0   
..     ...     ...     ...             ...            ...                ...   
271    272  Gentoo  Biscoe             NaN            NaN                NaN   
272    273  Gentoo  Biscoe            46.8           14.3              215.0   
273    274  Gentoo  Biscoe            50.4           15.7              222.0   
274    275  Gentoo  Biscoe            45.2           14.8              212.0   
275    276  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 9 columns]

In [18]:
chinstrap = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 1000)]

In [19]:
chinstrap

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [20]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              344 non-null    int64  
 1   species            344 non-null    object 
 2   island             344 non-null    object 
 3   bill_length_mm     342 non-null    float64
 4   bill_depth_mm      342 non-null    float64
 5   flipper_length_mm  342 non-null    float64
 6   body_mass_g        342 non-null    float64
 7   sex                333 non-null    object 
 8   year               344 non-null    int64  
dtypes: float64(4), int64(2), object(3)
memory usage: 24.3+ KB


In [21]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['year'] = penguins['year'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

In [22]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    int64   
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     342 non-null    float64 
 4   bill_depth_mm      342 non-null    float64 
 5   flipper_length_mm  342 non-null    float64 
 6   body_mass_g        342 non-null    float64 
 7   sex                333 non-null    category
 8   year               344 non-null    category
dtypes: category(4), float64(4), int64(1)
memory usage: 15.4 KB


<h1>FILLING UP NULL VALUES</h1>

In [23]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [24]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [25]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [26]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [27]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [28]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [29]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [30]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [31]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [32]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [33]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [34]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [35]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [36]:
penguins['sex'].mode()

0    male
Name: sex, dtype: category
Categories (2, object): ['female', 'male']

In [37]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [38]:
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [39]:
penguins.columns

Index(['rowid', 'species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [40]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000, 1)
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  
0           181.000000  3750.000000    male  2007      3.8  
1           186.000000  3800.000000  female  2007      3.8  
2           195.000000  3250.000000  female  2007      3.2  
3           200.915205  4201.754386    male  2007      4.2  
4           193.000000  3450.000000  female  2007      3.4  
..                 ...          ...     ...   ...      ...  
339         207.000000  4000.000000    male  2009      4.0  
340         202.000000  3400.000000  female  2009      3.4  
341         193.000000  3775.000000    male  2009      3.8  
342         210.000000  4100.000000    male  2009      4.1  
343         198.000000  3775.000000  female  2009      3.8  

[344 rows x 10 columns]

<h1>Questions</h1>

<hr>
<h1>Q1</h1>
<h2>Which penguin species has the highest average body mass?</h2>

In [41]:
penguins.groupby("species", observed=True)["body_mass_g"].mean().sort_values(ascending=False)

species
Gentoo       5068.965761
Chinstrap    3733.088235
Adelie       3703.958910
Name: body_mass_g, dtype: float64

<h2>INSIGHT</h2>
<h3>The highest average body mass is Gentoo.</h3>

<hr>
<h1>Q2</h1>
<h2>Which island has the most penguin observations?</h2>

In [42]:
penguins["island"].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h2>INSIGHT</h2>
<h3>The island have the most penguin observation is Biscoe.</h3>

<hr>
<h1>Q3</h1>
<h2>What are the average bill length and depth for each species?</h2>

In [43]:
penguins.groupby("species", observed=True)[["bill_length_mm", "bill_depth_mm"]].mean()

bill_length_mm  bill_depth_mm
species                                 
Adelie          38.825144      18.338495
Chinstrap       48.833824      18.420588
Gentoo          47.475983      14.999606

<h2>INSIGHT</h2>
<h3>The top penguin bill length and depth is Adelie.</h3>

<hr>
<h1>Q4</h1>
<h2>Which species has the longest average flipper length?</h2>

In [44]:
penguins.groupby("species", observed=True)["flipper_length_mm"].mean().sort_values(ascending=False)


species
Gentoo       217.055768
Chinstrap    195.823529
Adelie       190.025758
Name: flipper_length_mm, dtype: float64

<h2>INSIGHT</h2>
<h3>The longer average flipper length is Gentoo.</h3>

<hr>
<h1>Q5</h1>
<h2>Do male penguins weight more than female penguins?</h2>

In [45]:
penguins.groupby("sex", observed=True)["body_mass_g"].mean()

sex
female    3862.272727
male      4514.684407
Name: body_mass_g, dtype: float64

<h2>INSIGHT</h2>
<h3>So female was heavier than male penguins.</h3>

<hr>
<h1>Q6</h1>
<h2>What is the average body mass of each species per island?</h2>

In [46]:
island = penguins.groupby(["island", "species"], observed=True)["body_mass_g"].mean()
island

island     species  
Biscoe     Adelie       3709.659091
           Gentoo       5068.965761
Dream      Adelie       3688.392857
           Chinstrap    3733.088235
Torgersen  Adelie       3715.899123
Name: body_mass_g, dtype: float64

<h2>INSIGHT</h2>
<h3>This reveals that in biscoe have the most heaviest body mass species.</h3>

<hr>
<h1>Q7</h1>
<h2>How many penguins were observed each year?</h2>

In [47]:
penguins["year"].value_counts().sort_index()

year
2007    110
2008    114
2009    120
Name: count, dtype: int64

<h2>INSIGHT</h2>
<h3>The year have the most observed penguins was 2007.</h3>

In [48]:
island = penguins.groupby(["island", "species"], observed=True)["body_mass_g"].mean()
island

island     species  
Biscoe     Adelie       3709.659091
           Gentoo       5068.965761
Dream      Adelie       3688.392857
           Chinstrap    3733.088235
Torgersen  Adelie       3715.899123
Name: body_mass_g, dtype: float64

In [49]:
specie = penguins.groupby(["species","island"],observed = True)
specie

In [50]:
specie

In [51]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  
0           181.000000  3750.000000    male  2007      3.8  
1           186.000000  3800.000000  female  2007      3.8  
2           195.000000  3250.000000  female  2007      3.2  
3           200.915205  4201.754386    male  2007      4.2  
4           193.000000  3450.000000  female  2007      3.4  
..                 ...          ...     ...   ...      ...  
339         207.000000  4000.000000    male  2009      4.0  
340         202.000000  3400.000000  female  2009      3.4  
341         193.000000  3775.000000    male  2009      3.8  
342         210.000000  4100.000000    male  2009      4.1  
343         198.000000  3775.000000  female  2009      3.8  

[344 rows x 10 columns]

<hr>
<h1>Q8</h1>
<h2>Is there a correlation between the mass of the penguin and thier bill length</h2>

In [52]:
weyt_flip_cor = penguins["body_mass_g"].corr(penguins["bill_length_mm"])

In [53]:
weyt_flip_cor

np.float64(0.5951098244376305)

<h2>INSIGHT</h2>
<h3>There is a correlation between the body mass of the penguin and their bill length.</h3>

<hr>
<h1>Q9</h1>
<h2>How many male Adelie during Year 2007</h2>

In [54]:
male_df = penguins[penguins["species"]== "Adelie"]

In [55]:
male_df = penguins[penguins["year"]=='2007']

In [56]:
male_df = penguins[penguins["sex"]=='male']

In [57]:
male_df['sex'].value_counts()

sex
male      179
female      0
Name: count, dtype: int64

<h1>INSIGHT</h1>
<h2>There are 179 male Adelie during year 2007</h2>

<hr>
<h1>Q10</h1>
<h2>How many female Adelie during Year 2007</h2>

In [58]:
female_df = penguins[penguins["species"]== "Adelie"]

In [59]:
female_df = penguins[penguins["year"]=='2007']

In [60]:
female_df = penguins[penguins["sex"]=='female']

In [61]:
female_df['sex'].value_counts()

sex
female    165
male        0
Name: count, dtype: int64

<h1>INSIGHT</h1>
<h2>There are 165 female Adelie during year 2007</h2>

<hr>
<h1>Q11</h1>
<h2>Is there a correlation between the sex of the Chinstrap penguin and there body mass</h2>

In [62]:
penguins[penguins["species"]=="Chinstrap"]['body_mass_g']

276    3500.0
277    3900.0
278    3650.0
279    3525.0
280    3725.0
        ...  
339    4000.0
340    3400.0
341    3775.0
342    4100.0
343    3775.0
Name: body_mass_g, Length: 68, dtype: float64

In [63]:
penguins[penguins['sex']=='male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [64]:
penguins[penguins['sex']=='female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [65]:
f_oneway(penguins[penguins['sex']=='male']['body_mass_g'],penguins[penguins['sex']=='female']['body_mass_g'])

F_onewayResult(statistic=np.float64(68.38476171974526), pvalue=np.float64(3.0226600487050246e-15))

<hr>
<h1>INSIGHT</h1>
<h2>Which is the most populated island</h2>

<hr>
<h1>Q12</h1>
<h2>Is there a correlation between the island and it's total population</h2>

In [74]:
island1_df = penguins[penguins['island'] == 'Torgersen']

In [75]:
island1_df.value_counts().sum()

np.int64(52)

In [68]:
island2_df = penguins[penguins['island'] == 'Dream']

In [69]:
island2_df.value_counts().sum()

np.int64(124)

In [70]:
island3_df = penguins[penguins['island'] == 'Biscoe']

In [71]:
island3_df.value_counts().sum()

np.int64(168)

<hr>
<h1>INSIGHT</h1>
<h2>The mos populated Island is Bicoe then Dream and lastly Torgersen</h2>